src/TypeMapping/MysqlTypeMapper.php

```diff
@@ -117,13 +117,17 @@ public function mapToPHPStanType(string $mysqlType, array $mysqlFlags, int $leng
             $phpstanType = IntegerRangeType::fromInterval(0, 2155);
         }
         // floats are detected as numerics in mysqli
-        if (\in_array(strtoupper($mysqlType), ['DOUBLE', 'NEWDECIMAL', 'REAL'], true)) {
+        if (\in_array(strtoupper($mysqlType), ['DOUBLE', 'REAL'], true)) {
             $phpstanType = new FloatType();
         }
 
         // fallbacks
         if (null === $phpstanType) {
             switch (strtoupper($mysqlType)) {
+                case 'DECIMAL':
+                case 'NEWDECIMAL':
+                    $phpstanType = new AccessoryNumericStringType();
```

Comment:
See line 161 how a numeric string is built. An accessory type cannot stand on irs own

Reply:
Yea, I've seen it, but instantiated it just like this and it seems to work ? 🙈

Reply:
It works for the tests but will not work in the bigger picture when the type needs to be combined with other logic within phpstan